### PR TITLE
move lower var declarations

### PR DESCRIPTION
**Pull request by Alexey-T**

### CODE REVIEW

File: src/RegExpr.pas [hunk 1]
```diff
@@ -1575,9 +1574,9 @@ class function TRegExpr.InvertCaseFunction(const Ch: REChar): REChar;
 
 procedure TRegExpr.SetExpression(const AStr: RegExprString);
 begin
-  if (AStr <> fExpression) or not fExprIsCompiled then
+  if (AStr <> fExpression) or not regexpCompiled then
```
Comment (andgineer) [inline src/RegExpr.pas]: in fact this is very strange looking - past form instead of passive

File: src/RegExpr.pas [hunk 2]
```diff
@@ -2679,19 +2672,24 @@ function TRegExpr.ParsePiece(var flagp: integer): PRegExprChar;
...
       PREBracesArg(AlignToInt(Result + REOpSz + RENextOffSz))^ := ABracesMin;
-      PREBracesArg(AlignToInt(Result + REOpSz + RENextOffSz + REBracesArgSz))^
-        := ABracesMax;
+      PREBracesArg(AlignToInt(Result + REOpSz + RENextOffSz + REBracesArgSz))^ := ABracesMax;
     end;
   end;
 
+var
+  op: REChar;
+  NonGreedyOp, NonGreedyCh: boolean; // ###0.940
```
Comment (andgineer) [inline src/RegExpr.pas]: Sorry I do not get it - somewhere it starts from capital letter, somewhere else this is cameCase.

File: src/RegExpr.pas [hunk 3]
```diff
@@ -3492,15 +3490,15 @@ function TRegExpr.regrepeat(p: PRegExprChar; AMax: integer): integer;
         ArrayIndex := FSubExprIndexes[Ord(opnd^)];
         if ArrayIndex < 0 then
           Exit;
-        sestart := startp[ArrayIndex];
-        if sestart = nil then
+        GrpStart := startp[ArrayIndex];
```
Comment (andgineer) [inline src/RegExpr.pas]: Well it's ok to rename subexpressions to groups.
But can we do that in all places? Now you have FSubExprIndexes and GrpStart

**Comment by andgineer** [comment]
I can merge it if you like it.
But don't you think this is inconsistent in some places?

File: src/RegExpr.pas (hunk 1 above)
Comment (Alexey-T) [reply src/RegExpr.pas]: renamed to regexpIsCompiled, ok.

File: src/RegExpr.pas (hunk 2 above)
Comment (Alexey-T) [reply src/RegExpr.pas]: NonGreedyOp / NonGreedyCh are your old names, i moved them lower to make them not visible by included funcs.

**Comment by andgineer** [comment]
Can we have some test for this (that 'foobar{3,blahblah' raises exception)?

File: src/RegExpr.pas (hunk 3 above)
Comment (Alexey-T) [reply src/RegExpr.pas]: renamed to GrpIndexes/ GrpCount, okay.

**Comment by Alexey-T** [comment]
> Can we have some test for this (that 'foobar{3,blahblah' raises exception)?

@andgineer Test cannot show ok for property Error raised, so i tested in test_dlg program: now bad braces give error.

**Comment by andgineer** [comment]
Well all tests as different soubroutines so we can have special subroutine for this test

**Comment by Alexey-T** [comment]
Don't know how to write this- if like this, we don't get error msg at all
```
procedure TTestRegexpr.TestBadBraces;
begin
  try
    CompileRE('d{');
    IsTrue('Not raised Error for braces', False);
  except
  end;

  try
    CompileRE('d{2,');
    IsTrue('Not raised Error for braces', False);
  except
  end;

  try
    CompileRE('d{}');
    IsTrue('Not raised Error for braces', False);
  except
  end;
end;
```

**Pull request merged**